### ISSUE TITLE
Close all

### ISSUE THREAD

**Issue by Davincible** [silentsignal/burp-json-jtree#1]
Hey I know this is a bit older already, but I really like the plugin, and it would be even more amazing if there would be a right-click option to minimize all and minimize all children. That would make exploring large json responses a lot easier. Would it be possible to add that?

**Comment by dnet** [comment]
Thanks for the feedback! Just to clarify, let me rephrase what I understood from your request and you can confirm or correct me:

 - **Minimize all** regardless of whichever node you open the menu from, all nodes are collapsed, thus even if you expand one manually after this operation, you'll only see the direct descendants but not "grandchildren" nodes
 - **Minimize all children** the recursive collapse described in the above specification only gets applied to the selected node and its descendants

**Comment by Davincible** [comment]
100% what I mean. Would make it easier to explore large json structures, as currently, you need to collapse/close a single node manually, if one has 100 child nodes that can be a bit cumbersome.

**Comment by dnet** [comment]
Thanks for the suggestion, I released [v0.6](https://github.com/silentsignal/burp-json-jtree/releases/tag/v0.6) that includes this

**Comment by Davincible** [comment]
I've just tried it out, but I now get an ant (`version 1.10.10`) build error:
```
Buildfile: /home/tyler/Cyber/BurpSuite/Plugins/burp-json-jtree/build.xml

classes:
    [javac] Compiling 1 source file to /home/tyler/Cyber/BurpSuite/Plugins/burp-json-jtree/build
    [javac] /home/tyler/Cyber/BurpSuite/Plugins/burp-json-jtree/src/burp/JsonJTree.java:116: error: incompatible types: inference variable T has incompatible equality constraints DefaultMutableTreeNode,TreeNode
    [javac]             final ArrayList<DefaultMutableTreeNode> list = Collections.list(node.children());
    [javac]                                                                            ^
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in method <T>list(Enumeration<T>)
    [javac] Note: /home/tyler/Cyber/BurpSuite/Plugins/burp-json-jtree/src/burp/JsonJTree.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 1 error

BUILD FAILED
/home/tyler/Cyber/BurpSuite/Plugins/burp-json-jtree/build.xml:14: Compile failed; see the compiler error output for details.

Total time: 0 seconds

```

**Comment by dnet** [comment]
Interesting, `javac` from OpenJDK 8 and OpenJDK 11 compiles it fine for me. I attempted to replace generics with casting in a separate branch called `children-cast`, the commit ID is 8de9e5e. Could you check whether this fixes the issue in your environment?

**Comment by Davincible** [comment]
That does seem to work :)
